app/services/catalog/update_order_item.rb

```diff
@@ -0,0 +1,33 @@
+module Catalog
+  class UpdateOrderItem
+    class OrderItemNotFound < StandardError; end
+
+    def initialize(topic)
+      @payload = topic.payload
+      @message = topic.message
+    end
+
+    def process
+      Rails.logger.info("Processing service order topic message: #{@message} with payload: #{@payload}")
+
+      Rails.logger.info("Searching for OrderItem with a task_ref: #{@payload["task_id"]}")
+      order_item = OrderItem.where(:topology_task_ref => @payload["task_id"]).first
```

Comment:
@eclarizio Should the fetch of order_item, raising an exception be in a separate private function.